### PR TITLE
[com_fields] Rename the fields property to jcfields

### PR DESCRIPTION
**Pull request by laoneo**
Pull Request for Issue #14437.

### Summary of Changes
Renames the property fields on the `onContentPrepare` event to something more unique `jcfields`.

### Testing Instructions
Just create some custom fields for articles and open the article on the front.
Additionally you can in the file components/com_content/views/article/tmpl/default.php add on line 25 the following code snippet:
`print_r($this->item);die;`

### Expected result
All should work as expected, especially for 3rd party components which do have a fields property on their entities.

For the print_r test, an object with is printed which contains a property jcfields which contains the custom fields.

### Actual result
3rd party components which do have a fields property on their entities got changed to the com_fields.

An object with is printed which contains a property fields which contains the custom fields.

### Documentation Changes Required
If the developer documentation is written, then it needs to be changed that the property fields will be renamed to jcfields.

### CODE REVIEW

File: plugins/system/fields/fields.php [hunk 1]
```diff
@@ -364,10 +364,8 @@ public function onContentPrepare($context, $item)
```
Comment (b2z) [inline plugins/system/fields/fields.php]: `jcfields` here? ;)

**Comment by laoneo** [comment]
You'r right, fixed.

File: plugins/system/fields/fields.php (hunk 1 above)
Comment (Bakual) [inline plugins/system/fields/fields.php]: I wouldn't change the name of the internal variables. Only the property needs renaming.

**Comment by Bakual** [comment]
I think that has to be changed also in the rendering part:
https://github.com/joomla/joomla-cms/blob/staging/components/com_fields/layouts/fields/render.php#L44
and the special handling in contacts:
https://github.com/joomla/joomla-cms/blob/staging/components/com_contact/layouts/fields/render.php#L44
https://github.com/joomla/joomla-cms/blob/staging/components/com_contact/views/contact/view.html.php#L272-L275
https://github.com/joomla/joomla-cms/blob/staging/components/com_contact/views/contact/tmpl/default_user_custom_fields.php#L23
https://github.com/joomla/joomla-cms/blob/staging/templates/beez3/html/com_contact/contact/default_user_custom_fields.php#L22

And also in the search plugin:
https://github.com/joomla/joomla-cms/blob/staging/plugins/search/content/content.php#L328

There may be more places. That's just the ones I found while doing a search for "->fields "

**Comment by laoneo** [comment]
Thanks @Bakual, changed them.

**Comment by b2z** [comment]
@laoneo  as you have changed more files **Testing Instructions** should  be changed?

**Comment by laoneo** [comment]
True, changed it.

**Comment by b2z** [comment]
Applied the patch - fields are not being saved with article :(

**Comment by laoneo** [comment]
Did you edit on the front end?

**Comment by b2z** [comment]
Nope in back-end.

**Comment by laoneo** [comment]
Really strange, are you sure you are working on the latest staging? Did you get any error?

**Comment by b2z** [comment]
Opps, I was testing on beta3. No errors, I will try on staging.

**Comment by b2z** [comment]
I have tested this item :white_check_mark: successfully on d310fe397c9b601296c1b28b6e1b362ff3a1b453<br /><br />Content and contact fields are ok.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14493">issues.joomla.org/tracker/joomla-cms/14493</a>.</sub>

**Comment by b2z** [comment]
@ggppdk please help to test. Thanks!

**Comment by b2z** [comment]
@laoneo shlould it be Joomla! 3.7 milestone? It's a potential B/C break for 3rd party extensions.

**Comment by laoneo** [comment]
Why, com_fields was not released, so we can't break bc.

**Comment by b2z** [comment]
@laoneo ok let me rephrase - is it critical for 3.7 release or not? I mean breaking 3rd party extensions. Or 3rd party extensions must re-implement their code to support 3.7?

**Comment by Bakual** [comment]
This should go into 3.7.0. Afterwards it would only be possible for 4.0 since it would break implementations

**Comment by laoneo** [comment]
@b2z now I got it. Yes after the release, we can't change it before 4 anymore, even then I think ti would be not so a good idea. So yes this should go into 3.7. Hope we find one more tester.

**Comment by franz-wohlkoenig** [comment]
@laoneo don't understand test instructions like

> For the print_r test, an object with is printed which contains a property jcfields which contains the custom fields.

ar above my skills. 

If there is a 3rd-Party-Component to install i can test.

**Comment by laoneo** [comment]
That's mainly just a test that the field has the right name which will be used for extensions. More important are that the rest is working as expected.

**Comment by joomdonation** [comment]
I have tested this item :white_check_mark: successfully on d310fe397c9b601296c1b28b6e1b362ff3a1b453<br /><br />Works as expected. Tested with article and contact page<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14493">issues.joomla.org/tracker/joomla-cms/14493</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
RTC as there are 2 successfully Tests?

**Comment by jeckodevelopment** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14493">issues.joomla.org/tracker/joomla-cms/14493</a>.</sub>
